### ISSUE TITLE
Use EmitTypedEvent instead of EmitEvent which is deprecated

### ISSUE THREAD

**Issue by bsrinivas8687** [tendermint/liquidity#353]
## Summary

In the latest version of Cosmos SDK (current v0.42.x) type `EmitEvent` is deprecated so it's better go with `EmitTypedEvent`

This will be a breaking change for the clients which depends on events

Thoughts?

## Proposal

____

#### For Admin Use

- [x] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by kogisin** [comment]
Yes. We are aware of the fact that `EmitEvent` and `EmitEvents` will be deprecated soon. Since we're using Cosmos SDK v0.42.4 for the latest `develop` branch, we decided to use them for now.
